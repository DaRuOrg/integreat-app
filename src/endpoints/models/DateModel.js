export default class DateModel {
  constructor ({ startDate, startTime = '00:00:00', endDate, endTime, allDay = false }) {
    this._startDate = startDate
    this._startTime = startTime
    this._endDate = endDate
    this._endTime = endTime
    this._allDay = allDay
  }

  get startDate () {
    return this._startDate
  }

  get startTime () {
    return this._startTime
  }

  get endDate () {
    return this._endDate
  }

  get endTime () {
    return this._endTime
  }

  get allDay () {
    return this._allDay
  }

  static toLocaleDateTime (value, locale) {
    const locales = [ locale, 'en', 'de' ]
    const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric', hour: '2-digit', minute: '2-digit' }
    return new Date(value).toLocaleString(locales, options)
  }

  static toLocaleDate (value, locale) {
    const locales = [ locale, 'en', 'de' ]
    const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' }
    return new Date(value).toLocaleDateString(locales, options)
  }

  static toLocaleTime (value, locale) {
    const locales = [ locale, 'en', 'de' ]
    const options = { hour: '2-digit', minute: '2-digit' }
    return new Date(value).toLocaleTimeString(locales, options)
  }

  toLocaleString (locale) {
    const oClock = locale === 'de' ? ' Uhr' : ''
    if (this.allDay !== '0') {
      if (this.endDate && this.endDate !== this.startDate) {
        return DateModel.toLocaleDate(this.startDate, locale) + ' - ' + this.toLocaleDate(this.endDate, locale)
      } else {
        return DateModel.toLocaleDate(this.startDate, locale)
      }
    } else {
      if (this.endDate && this.endDate !== this.startDate) {
        return DateModel.toLocaleDateTime(this.startDate + ' ' + this.startTime, locale) + oClock +
          ' - ' + DateModel.toLocaleDateTime(this.endDate + ' ' + this.endTime) + oClock
      } else if (this.endDate === this.startDate && this.endTime !== this.startTime) {
        return DateModel.toLocaleDateTime(this.startDate + ' ' + this.startTime, locale) + ' - ' +
          DateModel.toLocaleTime(this.endDate + ' ' + this.endTime, locale) + oClock
      } else if (DateModel.startTime) {
        return DateModel.toLocaleDateTime(this.startDate + ' ' + this.startTime, locale) + oClock
      } else {
        return DateModel.toLocaleDate(this.startDate, locale)
      }
    }
  }
}
