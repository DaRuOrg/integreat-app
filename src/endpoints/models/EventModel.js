export default class EventModel {
  constructor ({ id, title = '', content = '', thumbnail = '', address = '', town = '', date = {} }) {
    this._id = id
    this._title = title
    this._content = content
    this._thumbnail = thumbnail
    this._address = address
    this._town = town
    this._date = date
  }

  get id () {
    return this._id
  }

  get title () {
    return this._title
  }

  get thumbnail () {
    return this._thumbnail
  }

  get content () {
    return this._content
  }

  get address () {
    return this._address + ', ' + this._town
  }

  getDate (locale) {
    return this._date.toLocaleString(locale)
  }
}
