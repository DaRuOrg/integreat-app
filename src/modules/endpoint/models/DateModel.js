class DateModel {
  constructor ({ startDate, endDate, allDay }) {
    this._startDate = startDate
    this._endDate = endDate
    this._allDay = allDay
  }

  /**
  * (!) startDate and endDate are not in German, but UTC-Timezone. They are sent in German timezone by Server, but without any
  * timezone declaration, so we don't know if it's DaylightSavingTime or not. So we parse it in UTC and display it in UTC.
  * @return {Date} the Date when the event ends. If it's not specified, then null. Must be displayed in UTC-Timezone.
  */
  get startDate () {
    return this._startDate
  }

  /**
  * @return {Date} the Date when the event starts. Must be displayed in UTC-Timezone.
  */
  get endDate () {
    return this._endDate
  }

  /** @return {boolean} true, if event is all day long. */
  get allDay () {
    return this._allDay
  }
}

export default DateModel
