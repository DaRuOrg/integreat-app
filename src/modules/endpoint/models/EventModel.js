// @flow

import moment from 'moment'

type eventModel = {
  id: boolean,
  title: string,
  content: string,
  thumbnail: string,
  availableLanguages: any,
  content: string,
  town: string,
  address: string,
  startDate: moment,
  endDate: moment,
  allDay: boolean,
  excerpt: string
}

class EventModel {
  _id: boolean
  _title: string
  _content: string
  _thumbnail: string
  _availableLanguages: any
  _content: string
  _town: string
  _address: string
  _startDate: moment
  _endDate: moment
  _allDay: boolean
  _excerpt: string

  constructor (eventModel: eventModel) {
    this._id = eventModel.id
    this._title = eventModel.title
    this._content = eventModel.content
    this._thumbnail = eventModel.thumbnail
    this._address = eventModel.address
    this._town = eventModel.town
    this._startDate = eventModel.startDate
    this._endDate = eventModel.endDate
    this._allDay = eventModel.allDay
    this._excerpt = eventModel.excerpt
    this._availableLanguages = eventModel.availableLanguages
  }

  get id (): boolean {
    return this._id
  }

  get title (): string {
    return this._title
  }

  get thumbnail (): string {
    return this._thumbnail
  }

  get content (): string {
    return this._content
  }

  get address (): ?string {
    if (this._address && this._town) {
      return `${this._address}, ${this._town}`
    } else if (this._address) {
      return this._address
    } else if (this._town) {
      return this._town
    }
  }

  get startDate (): moment {
    return this._startDate
  }

  get endDate (): moment {
    return this._endDate
  }

  get allDay (): boolean {
    return this._allDay
  }

  get excerpt (): string {
    return this._excerpt
  }

  get availableLanguages (): any {
    return this._availableLanguages
  }
}

export default EventModel
