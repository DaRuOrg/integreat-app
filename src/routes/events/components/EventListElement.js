import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'redux-little-router'

import cx from 'classnames'

import EventModel from '../../../modules/endpoint/models/EventModel'

import style from './EventListElement.css'
import EventPlaceholder1 from '../assets/EventPlaceholder1.jpg'
import EventPlaceholder2 from '../assets/EventPlaceholder2.jpg'
import EventPlaceholder3 from '../assets/EventPlaceholder3.jpg'
import RemoteContent from 'modules/common/components/RemoteContent'

class EventListElement extends React.Component {
  static propTypes = {
    event: PropTypes.instanceOf(EventModel).isRequired,
    parentUrl: PropTypes.string.isRequired,
    language: PropTypes.string.isRequired,
    isFirst: PropTypes.bool.isRequired
  }

  getUrl () {
    return `${this.props.parentUrl}/${this.props.event.id}`
  }

  getEventPlaceholder () {
    const placeholders = [EventPlaceholder1, EventPlaceholder2, EventPlaceholder3]
    return placeholders[this.props.event.id % 3]
  }

  render () {
    return (
      <Link href={this.getUrl()} className={this.props.isFirst ? cx(style.firstEvent, style.event) : style.event}>
        <img className={style.eventThumbnail} src={this.props.event.thumbnail || this.getEventPlaceholder()}/>
        <div className={style.eventDescription}>
          <div className={style.eventTitle}>{this.props.event.title}</div>
          <div className={style.eventDate}>{this.props.event.getDate(this.props.language)}, {this.props.event.address}</div>
          <RemoteContent dangerouslySetInnerHTML={{__html: this.formatExcerpt()}}/>
        </div>
      </Link>
    )
  }

  formatExcerpt () {
    return this.props.event.excerpt.slice(0, 70) + '...'
  }
}

export default EventListElement
