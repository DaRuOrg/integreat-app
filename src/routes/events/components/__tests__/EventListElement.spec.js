import React from 'react'
import { shallow } from 'enzyme'

import EventModel from 'modules/endpoint/models/EventModel'
import EventListElement from '../EventListElement'
import DateModel from '../../../../modules/endpoint/models/DateModel'

describe('EventDetail', () => {
  const event = new EventModel({
    id: 1234,
    title: 'first Event',
    availableLanguages: {de: '1235', ar: '1236'},
    date: new DateModel({startDate: new Date('2017-11-18' + 'T' + '09:30:00' + 'Z'),
      endDate: new Date('2017-11-18' + 'T' + '19:30:00' + 'Z'),
      allDay: true})

  })

  const url = `/augsburg/en/events`

  const language = 'en'

  test('should render', () => {
    expect(shallow(
      <EventListElement event={event} parentUrl={url} isFirst={false} language={language} />
    )).toMatchSnapshot()
  })
})
