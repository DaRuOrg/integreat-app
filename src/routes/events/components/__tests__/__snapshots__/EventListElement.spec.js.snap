// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetail should render 1`] = `
<Connect(Link)
  className="event"
  href="/augsburg/en/events/1234"
>
  <img
    className="eventThumbnail"
    src="test-file-stub"
  />
  <div
    className="eventDescription"
  >
    <div
      className="eventTitle"
    >
      first Event
    </div>
    <div
      className="eventDate"
    >
      <Timespan
        locale="en"
      />
      , 
    </div>
    <RemoteContent
      dangerouslySetInnerHTML={
        Object {
          "__html": "...",
        }
      }
    />
  </div>
</Connect(Link)>
`;
